### PR TITLE
Support null/undefined table cells

### PR DESCRIPTION
**Pull request by mykolaharmash**
## Purpose

Right now <Table> crashes in case `null` or `undefined` were provider as cell content. This change makes it just render an empty string.

## Approach and changes

* Extending `Cell` type definition
* Making sure methods that use cell's content expect void values

### CODE REVIEW

**Comment by mykolaharmash** [comment]
Hmm, the login button on the Contributor License Agreement page does not seem to work.

https://user-images.githubusercontent.com/793851/117445060-8b011500-af3a-11eb-9a42-9c004dffa18f.mp4

In dev tools see 401 from `https://clark-a52pucifqa-ez.a.run.app/api/auth/authenticate`.

File: packages/circuit-ui/components/Table/utils.ts [hunk 1]
```diff
@@ -23,17 +23,21 @@ export const mapRowProps = (props: Row): { cells: Cell[] } =>
...
 
-export const getSortByValue = (props: Cell): SortByValue | ReactNode =>
-  typeof props === 'object' && props.sortByValue !== undefined
-    ? props.sortByValue
-    : getCellChildren(props);
+export const getSortByValue = (props: Cell): SortByValue | ReactNode => {
+  const cell = mapCellProps(props);
+
+  return cell.sortByValue !== undefined ? cell.sortByValue : cell.children;
+};
```
Comment (robinmetral) [inline packages/circuit-ui/components/Table/utils.ts:40]: Nice, do you think we could even clean up this `getCellChildren` altogether or is it used in too many places?

**Review by robinmetral — COMMENTED** [review]
🎉 thanks for your contribution!

Could you add a test case for a null or undefined cell?

File: packages/circuit-ui/components/Table/types.ts [hunk 2]
```diff
@@ -25,7 +25,7 @@ export type CellObject = {
   sortByValue?: SortByValue;
 };
 
-export type Cell = string | number | CellObject;
+export type Cell = string | number | CellObject | null | undefined;
```
[the remark this answers is not in the document]
Comment (robinmetral) [reply packages/circuit-ui/components/Table/types.ts:28]: Actually there's probably a good case for `undefined`, especially since Tables are usually populated with data from API responses, where undefined is plausible

**Comment by robinmetral** [comment]
I forgot something 🙈 Could you also add changesets for this fix? You can use `yarn changesets` for it and push the files it generates (more [here](https://circuit.sumup.com/?path=/docs/introduction-contributing-release-process--page)).

File: packages/circuit-ui/components/Table/types.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mykolaharmash) [reply packages/circuit-ui/components/Table/types.ts:28]: Right, that was my logic as well, to support both common void values.

**Comment by mykolaharmash** [comment]
Added the test, changeset and rebased on `main`.

**Review by robinmetral — APPROVED** [review]
🙌
